src/shared/start-gateway.js

```diff
@@ -83,6 +83,45 @@ export async function startGateway<
         logger.info(`${name} running at http://${host}:${port}`);
     });
 
+    /**
+     * In this server is being run using a process manager, such as PM2, we
+     * may be asked to shutdown gracefully. We do this be listening for the
+     * SIGINT signal and then close the server. This prevents new connections
+     * from coming in and waits until the existing connections complete before
+     * the callback is fired. At which point we can safely shutdown the server.
+     * If we fail to respond then the process manager may try to forcefully
+     * shutdown the server after a timeout.
+     */
+    process.on("SIGINT", () => {
+        if (!gateway) {
+            return;
+        }
+
+        logger.info("SIGINT received, shutting down server.");
+
+        try {
+            gateway.close((err) => {
+                if (err) {
+                    logger.error(
+                        `Error shutting down server: ${
+                            (err && err.message) || "Unknown Error"
+                        }`,
+                    );
+                    process.exit(1);
```

Comment:
Could you move the `process.exit(1)` to the end of the method and then you wouldn't need it in two places? It would just be the default.

Reply:
I don't think that would work unfortunately - since one is in a callback and one happens synchronously...

Reply:
(to be clear, I tried to make it work but I just ended up with the code duplication anyway)

Reply:
Ah yeah, you're right. I missed that bit. Doh! Thanks for taking a look :)